apps/wallet/src/background/account-sources/index.ts

```diff
@@ -72,6 +72,32 @@ async function createAccountSource({
 	}
 }
 
+async function deleteAccountSourceByType({
+	type,
+}: MethodPayload<'deleteAccountSourceByType'>['args']) {
+	switch (type) {
+		case 'mnemonic':
+			try {
+				const sourceIds = await (await getDB()).accountSources
+					.where('type')
+					.anyOf(['mnemonic', 'imported'])
+					.primaryKeys();
+				const accountIds = await (await getDB()).accounts
+					.where('sourceID')
+					.anyOf(sourceIds)
+					.primaryKeys();
+				await (await getDB()).accounts.bulkDelete(accountIds);
+				await (await getDB()).accountSources.bulkDelete(accountIds);
+				return true;
```

Comment:
We probably need to notify dapps/update permissions because maybe some saved permissions might use the deleted accounts. I think emitting `accountsChanged` event here might work. But maybe because we create the new account after this it might be done there.